### PR TITLE
[Issue 5401][docs] Add deploy-docker doc

### LINKED ISSUE #5401: Create documentation for docker deployment on bare metal
**Is your feature request related to a problem? Please describe.**
I'd like to setup ZooKeeper, BookKeeper and Pulsar cluster on bare metal server not on k8s, using official docker image provided by Pulsar project, but currently there is not a documentation about this scenario.
So I have to decipher how to use official docker image from k8s config.
On bare metal setup, I prefer docker deployment to manual tarball installation because some operational benefit(e.g. easy to upgrade).
I also prefer official docker image to home grown one even for ZooKeeper and BookKeeper cluster because it should have accumulated community knowledge.

**Describe the solution you'd like**
It is very helpful if there will be a documentation for docker deployment on bare metal: how to use official docker image to setup ZooKeeper, BookKeeper and Pulsar cluster.

**Describe alternatives you've considered**

**Additional context**
I consulted mail list about this issue and was suggested raising issue.
http://mail-archives.apache.org/mod_mbox/pulsar-users/201910.mbox/browser

### PR DESCRIPTION
**Pull request by sijia-w**
Fixes #5401

Motivation:
Add deployment on Docker doc.

Modification:
The doc about how to deploy a cluster on Docker is added and the sidebar is updated accordingly.

### CODE REVIEW

**Comment by sijia-w** [comment]
@zymap @Jennifer88huang Please help review the added doc.

File: site2/docs/deploy-docker.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+```
+docker network connect pulsar zookeeper
+docker network connect pulsar bookkeeper
+docker network connect pulsar broker
+```
+
+To check whether the containers are successfully connected to the network, enter the `docker network inspect pulsar` command.
+
+For detailed information about deployment for each Pulsar component, see [deploy a cluster on bare metal](https://pulsar.apache.org/docs/en/deploy-bare-metal/).
```
Comment (Jennifer88huang) [inline site2/docs/deploy-docker.md:51]:
```suggestion
For detailed information about how to deploy ZooKeeper cluster, BookKeeper cluster, brokers, see [deploy a cluster on bare metal](deploy-bare-metal.md).
```

Comment (Jennifer88huang) [reply site2/docs/deploy-docker.md:51]: 1. When you tell users Pulsar component, they might think which component.
2. Use relative path in reference link.

**Comment by Jennifer88huang** [comment]
@sijia-w you can rename the PR title and its description specified in the PR template.

**Comment by sijia-w** [comment]
@Jennifer88huang Thank you for your comments and help.

**Comment by Jennifer88huang** [comment]
@zymap Do you have any comments on this PR?

**Pull request merged**